h2o-py/h2o/explanation/_explain.py

```diff
@@ -1305,35 +1308,59 @@ def ice_plot(
             factor_map = _factor_mapper(NumpyFrame(frame[column]).from_factor_to_num(column))
 
         deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
+        deciles[10] = frame.nrow - 1
         colors = plt.get_cmap(colormap, 11)(list(range(11)))
         plt.figure(figsize=figsize)
         for i, index in enumerate(deciles):
+            percentile_string = "{}th Percentile".format(i * 10)
             tmp = NumpyFrame(
                 model.partial_plot(
                     frame,
                     cols=[column],
                     plot=False,
                     row_index=index,
                     targets=target,
-                    nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
+                    nbins=100 if not is_factor else 1 + frame[column].nlevels()[0],
+                    include_na=True
                 )[0]
             )
+            user_splits = dict()
+            orig_value = frame.as_data_frame()[column][index]
             encoded_col = tmp.columns[0]
+            if _isnan(orig_value):
```

Comment:
nit/style: This whole if block is pretty hard to parse, please consider separating it into a separate function. I'd also suggest adding some unit tests for it, but I'm not sure if you usually write units for such cases, so consider this optional 🙂

Reply:
added test_handle_orig_values() to pyunit_ice_plot.py